plans/dht/utils/staging.go

```diff
@@ -69,7 +69,7 @@ func (s *BatchStager) End() error {
 
 	t = time.Now()
 
-	err = <-s.ri.Watcher.Barrier(s.ctx, stage, int64(s.total-1))
+	err = <-s.ri.Client.MustBarrier(s.ctx, stage, s.total-1).C
```

Comment:
this can probably be `s.total` now, I was running into some issues before, but I suspect that with the refactor it should work fine now.

Reply:
Done.